src/components/app-profile/app-profile.e2e.ts

```diff
@@ -0,0 +1,27 @@
+import { newE2EPage } from '@stencil/core/testing';
+
+describe('app-profile', () => {
+  it('renders', async () => {
+    const page = await newE2EPage();
+    await page.setContent('<app-profile></app-profile>');
+
+    const element = await page.find('app-profile');
+    expect(element).toHaveClass('hydrated');
+  });
+
+  it('displays the specified name', async () => {
+    const page = await newE2EPage({ url: '/profile/joseph' });
+
+    const profileElement = await page.find('app-root >>> app-profile');
+    const element = profileElement.shadowRoot.querySelector('div');
+    expect(element.textContent).toContain('Hello! My name is Joseph.');
+  });
+
+  // it('includes a div with the class "app-profile"', async () => {
+  //   const page = await newE2EPage({ url: '/profile/joseph' });
+
+  // I would like to use a selector like this above, but it does not seem to work
+  //   const element = await page.find('app-root >>> app-profile >>> div');
```

Comment:
I am not sure how to get deeply into shadow DOM items. See line 16 for one way I was successful, but I don't really like that so much.